msi/createmsi.py

```diff
@@ -0,0 +1,162 @@
+    <Package Id='*' Keywords='Installer' Description="Meson %s installer"
+             Comments='Meson is a high performance build system' Manufacturer='Meson development team'
+             InstallerVersion='100' Languages='1033' Compressed='yes' SummaryCodepage='1252' />
+
+    <Media Id="1" Cabinet="meson.cab" EmbedCab="yes" />
+
+    <Directory Id='TARGETDIR' Name='SourceDir'>
+      <Directory Id="ProgramFilesFolder">
+        <Directory Id="INSTALLDIR" Name="Meson">
+'''
+
+xml_footer_templ = '''
+        </Directory>
+      </Directory>
+    </Directory>
+
+    <Feature Id="DefaultFeature" Level="1">
+%s
+    </Feature>
+
+  <Property Id="WIXUI_INSTALLDIR" Value="INSTALLDIR" />
+  <UIRef Id="WixUI_InstallDir" />
+
+  </Product>
+</Wix>
+'''
+
+file_templ = '''<File Id='%s' Name='%s' DiskId='1' Source='%s'  />
+'''
+
+comp_ref_templ = '''<ComponentRef Id="%s" />
+'''
+
+path_addition_xml = '''<Environment Id="Environment" Name="PATH" Part="last" System="yes" Action="set" Value="[INSTALLDIR]"/>
+'''
+
+
+def gen_guid():
+    return str(uuid.uuid4()).upper()
+
+class Node:
+    def __init__(self, dirs, files):
+        assert(isinstance(dirs, list))
+        assert(isinstance(files, list))
+        self.dirs = dirs
+        self.files = files
+
+class PackageGenerator:
+
+    def __init__(self):
+        self.version = coredata.version.replace('dev', '')
+        self.guid = 'DF5B3ECA-4A31-43E3-8CE4-97FC8A97212E'
+        self.update_guid = '141527EE-E28A-4D14-97A4-92E6075D28B2'
+        self.main_xml = 'Meson.wxs'
+        self.main_o = 'Meson.wixobj'
+        self.bytesize = '32' if '32' in platform.architecture()[0] else '64'
+        self.final_output = 'meson-%s-%s.msi' % (self.version, self.bytesize)
+        self.staging_dir = 'dist'
+
+    def build_dist(self):
+        if os.path.exists(self.staging_dir):
+            shutil.rmtree(self.staging_dir)
+        modules = [os.path.splitext(os.path.split(x)[1])[0] for x in glob(os.path.join('mesonbuild/modules/*'))]
+        modules = ['mesonbuild.modules.' + x for x in modules if not x.startswith('_')]
+        modulestr = ','.join(modules)
+        subprocess.check_call(['c:\\Python\python.exe',
+                               'c:\\Python\Scripts\\cxfreeze',
+                               '--target-dir',
+                               self.staging_dir,
+                               '--include-modules',
+                               modulestr,
+                               'meson.py'])
+        shutil.copy(shutil.which('ninja'), self.staging_dir)
+        if not os.path.exists(os.path.join(self.staging_dir, 'meson.exe')):
+            sys.exit('Meson exe missing from staging dir.')
+        if not os.path.exists(os.path.join(self.staging_dir, 'ninja.exe')):
+            sys.exit('Ninja exe missing from staging dir.')
+        
+
+    def generate_files(self):
+        assert(os.path.isdir(self.staging_dir))
+        #cumulative_xml = ''
+        component_num = 0
```

Comment:
__[flake8]__
* [F841] local variable 'component_num' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->